packages/components/src/font-size-picker/index.tsx

```diff
@@ -133,53 +175,52 @@ const UnforwardedFontSizePicker = (
 					{ ! disableCustomFontSizes && (
 						<HeaderToggle
 							label={
-								showCustomValueControl
+								currentPickerType === 'custom'
 									? __( 'Use size preset' )
 									: __( 'Set custom size' )
 							}
 							icon={ settings }
 							onClick={ () => {
-								setShowCustomValueControl(
-									! showCustomValueControl
+								setCurrentPickerType(
+									getPickerType(
+										currentPickerType !== 'custom',
+										fontSizes
```

Comment:
If we moved the `getPickerType` and `getHeaderHint` inside the component function, we wouldn't need to pass `fontSizes` all over the place. We could also potentially reuse some of the conditions (like whether the current picker type is `custom` for example), or even remove some of the utilities (like `shouldUseSelectOverToggle()`). Maybe that could reduce some of the complexity?

Reply:
I had initially extracted those function to a separate utility to better isolate the key pieces of logic and make the render function easier to scan.

But after applying Lena's advice, I noticed that some extra code could be removed — and so I put them back in the render function:

- https://github.com/WordPress/gutenberg/pull/63553/commits/41deb9f7b62621cbe75dcd8e158963d0dec395f0
- https://github.com/WordPress/gutenberg/pull/63553/commits/c53036ad3df5f85ff4b817dd6de6e7123b1bb6d3